.taskcluster.yml

```diff
@@ -48,168 +44,210 @@ tasks:
+          then: ${event.action}
+          else: 'UNDEFINED'
+
       in:
         $match:
           "tasks_for == 'github-pull-request' && event['action'] in ['opened', 'reopened', 'edited', 'synchronize']":
             $let:
-              pull_request_title: ${event.pull_request.title}
-              pull_request_number: ${event.pull_request.number}
-              pull_request_url: ${event.pull_request.html_url}
+              level:
+                $if: 'tasks_for in ["github-push"] && repoUrl == "https://github.com/mozilla/glean"'
+                then: '3'
+                else: '1'
             in:
-              $mergeDeep:
-                - {$eval: 'default_task_definition'}
-                - payload:
-                    env:
-                      GITHUB_PR_TITLE: ${pull_request_title}
-                - metadata:
-                    name: 'Glean - Decision task (Pull Request #${pull_request_number})'
-                    description: 'Building and testing Glean - triggered by [#${pull_request_number}](${pull_request_url})'
+              taskId: '${ownTaskId}'
+              taskGroupId:
+                  $if: 'tasks_for == "action"'
+                  then:
+                      '${action.taskGroupId}'
+                  else:
+                      '${ownTaskId}'  # same as taskId; this is how automation identifies a decision task
+              schedulerId: '${trustDomain}-level-${level}'
+              created: {$fromNow: ''}
+              deadline: {$fromNow: '1 day'}
+              expires: {$fromNow: '1 year 1 second'}  # 1 second so artifacts expire first, despite rounding errors
+              metadata:
+                $merge:
+                  - owner: "${ownerEmail}"
+                    source: '${repoUrl}/raw/${head_sha}/.taskcluster.yml'
+                  - $if: 'tasks_for in ["github-push", "github-pull-request"]'
+                    then:
+                      name: "Decision Task"
+                      description: 'The task that creates all of the other tasks in the task graph'
+              provisionerId: "glean-${level}"
+              workerType: "decision"
+              tags:
+                kind: decision-task
+              routes:
+                - checks
+              scopes:
+                # `https://` is 8 characters so, ${repoUrl[8:]} is the repository without the protocol.
+                $if: 'tasks_for == "github-push"'
+                then:
+                  $let:
+                    short_head_branch:
+                      $if: 'head_branch[:11] == "refs/heads/"'
+                      then: {$eval: 'head_branch[11:]'}
+                      else: ${head_branch}
+                  in:
+                    - 'assume:repo:${repoUrl[8:]}:branch:${short_head_branch}'
+                else:
+                  $if: 'tasks_for == "github-pull-request"'
+                  then:
+                    - 'assume:repo:github.com/${event.pull_request.base.repo.full_name}:pull-request'
```

Comment:
I'll need to generate all these roles/scopes along the way, before we can make this work. Tracked in the same aforementioned bug.